modules/core/02-client/keeper/client_test.go

```diff
@@ -508,3 +508,135 @@ func (suite *KeeperTestSuite) TestUpdateClientEventEmission() {
 	}
 	suite.Require().True(contains)
 }
+
+func (suite *KeeperTestSuite) TestRecoverClient() {
```

Comment:
Is it possible to bring here all the existing tests in [`TestClientUpdateProposal`](https://github.com/cosmos/ibc-go/blob/4e0bea725d57e0eec2ea485d2e8aa3085bc7079f/modules/core/02-client/keeper/proposal_test.go#L13)?

Reply:
yea! some are renamed but got em all from what I can tell.

Reply:
wasn't checking what I believe was intended and uncaught due to using expPass. That test fails during the height check since the tendermint client has a revision number of 1 while solomachine hardcodes it to 0. The new test `"CheckSubstituteAndUpdateState fails, substitute client trust level doesn't match subject client trust level"` takes care of making `CheckSubstituteAndUpdateState` fail.

Reply:
Makes sense, I suppose each light client implementation should be in charge of making sure the reference client is valid